RecoTracker/LST/plugins/LSTPhase2OTHitsInputProducer.cc

```diff
@@ -0,0 +1,62 @@
+#include "FWCore/Framework/interface/global/EDProducer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+#include "RecoTracker/LST/interface/LSTPhase2OTHitsInput.h"
+
+class LSTPhase2OTHitsInputProducer : public edm::global::EDProducer<> {
+public:
+  explicit LSTPhase2OTHitsInputProducer(edm::ParameterSet const& iConfig);
+  ~LSTPhase2OTHitsInputProducer() override = default;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  void produce(edm::StreamID, edm::Event& iEvent, const edm::EventSetup& iSetup) const override;
+
+  const edm::EDGetTokenT<Phase2TrackerRecHit1DCollectionNew> phase2OTRecHitToken_;
+  const edm::EDPutTokenT<LSTPhase2OTHitsInput> lstPhase2OTHitsInputPutToken_;
+};
+
+LSTPhase2OTHitsInputProducer::LSTPhase2OTHitsInputProducer(edm::ParameterSet const& iConfig)
+    : phase2OTRecHitToken_(
+          consumes<Phase2TrackerRecHit1DCollectionNew>(iConfig.getParameter<edm::InputTag>("phase2OTRecHits"))),
+      lstPhase2OTHitsInputPutToken_(produces<LSTPhase2OTHitsInput>()) {}
+
+void LSTPhase2OTHitsInputProducer::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  edm::ParameterSetDescription desc;
+
+  desc.add<edm::InputTag>("phase2OTRecHits", edm::InputTag("siPhase2RecHits"));
+
+  descriptions.addWithDefaultLabel(desc);
+}
+
+void LSTPhase2OTHitsInputProducer::produce(edm::StreamID iID, edm::Event& iEvent, const edm::EventSetup& iSetup) const {
+  // Setup
+  auto const& phase2OTHits = iEvent.get(phase2OTRecHitToken_);
+
+  // Vector definitions
+  std::vector<unsigned int> ph2_detId;
+  std::vector<float> ph2_x;
+  std::vector<float> ph2_y;
+  std::vector<float> ph2_z;
+  std::vector<TrackingRecHit const*> ph2_hits;
```

Comment:
Comment from Andrea:
> I would suggest to pre-compute the final size and reserve() the vectors to avoid the multiple reallocations.

Do you see any easy way to precompute this? I understand that these vectors should be of size `phase2OTHits.size() * it->size()`, and that this is very much dependent on the occupancy of the tracker per event, which can vary by a lot. Printing out the numbers for 10 events:
```
phase2OTHits.size()
it->size()

2093
18

1625
25

2772
17

2077
24

1390
14

1867
28

2967
21

3667
22

3001
46

4158
18
```

Reply:
Thinking about this some more, I was wrong in the calculation of the size of the vectors. I corrected the calculation in the original comment above, and from that it follows that `ph2_hits.size()` >= `phase2OTHits.size()`. Running again for 10 events:
```
ph2_hits.size()

3243

2429

3811

2756

2043

2872

3839

5379

4424

5922
```

But are you saying that having the estimate `ph2_hits.reserve(phase2OTHits.size());` is good enough, so I should just implement that?

Reply:
I should've checked the interface; it should be  `ph2_hits.reserve(phase2OTHits.dataSize());`